target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-mf289f.dts

```diff
@@ -0,0 +1,433 @@
+                bootargs-append = " root=/dev/ubiblock0_1";
+        };
+
+        gpio-restart {
+                compatible = "gpio-restart";
+                gpios = <&tlmm 8 GPIO_ACTIVE_HIGH>;
+        };
+
+        leds {
+                compatible = "gpio-leds";
+
+                led_internal: led-0 {
+                        function = LED_FUNCTION_STATUS;
+                        color = <LED_COLOR_ID_BLUE>;
+                        gpios = <&tlmm 10 GPIO_ACTIVE_LOW>;
+                        label = "blue:internal_led";
+                        default-state = "keep";
+                };
+
+                led-1 {
+                        function = LED_FUNCTION_WLAN;
+                        color = <LED_COLOR_ID_BLUE>;
+                        gpios = <&tlmm 61 GPIO_ACTIVE_LOW>;
+                        linux,default-trigger = "phy0tpt";
+                };
+        };
+
+        keys {
+                compatible = "gpio-keys";
+
+                reset {
+                        label = "reset";
+                        linux,code = <KEY_RESTART>;
+                        gpios = <&tlmm 18 GPIO_ACTIVE_LOW>;
+                };
+
+                wps {
+                        label = "wps";
+                        linux,code = <KEY_WPS_BUTTON>;
+                        gpios = <&tlmm 68 GPIO_ACTIVE_LOW>;
+                };
+        };
+
+        soc {
+                rng@22000 {
+                        status = "okay";
+                };
+
+                mdio@90000 {
+                        status = "okay";
+                        pinctrl-0 = <&mdio_pins>;
+                        pinctrl-names = "default";
+                        reset-gpios = <&tlmm 47 GPIO_ACTIVE_LOW>;
+                        reset-delay-us = <2000>;
+                };
+
+                ess-psgmii@98000 {
+                        status = "okay";
+                };
+
+                tcsr@1949000 {
+                        compatible = "qcom,tcsr";
+                        reg = <0x1949000 0x100>;
+                        qcom,wifi_glb_cfg = <TCSR_WIFI_GLB_CFG>;
+                };
+
+                tcsr@194b000 {
+                        /* select hostmode */
+                        compatible = "qcom,tcsr";
+                        reg = <0x194b000 0x100>;
+                        qcom,usb-hsphy-mode-select = <TCSR_USB_HSPHY_HOST_MODE>;
+                        status = "okay";
+                };
+
+                ess_tcsr@1953000 {
+                        compatible = "qcom,tcsr";
+                        reg = <0x1953000 0x1000>;
+                        qcom,ess-interface-select = <TCSR_ESS_PSGMII>;
+                };
+
+                tcsr@1957000 {
+                        compatible = "qcom,tcsr";
+                        reg = <0x1957000 0x100>;
+                        qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
+                };
+
+                usb2@60f8800 {
+                        status = "okay";

```

Comment:
On MF286D this was attached to external USB A port, is this controller used on the device, beside USB3 being used for internal modem?

Reply:
I see an unpopulated Micro-USB socket on the board photos from FCC documents, so please disregard the above.